### DISCUSSION TITLE
Execute electron api's while the app is only opened in the system tray

### DISCUSSION THREAD

**Discussion by MohammedChaibi** [webdriverio-community/wdio-electron-service#426] [Q&A]
Hi!

I'm new to Automated testing so I was hoping somebody could clarify this to me.

I have an electron app that only opens in the system tray, and in my test I'm hoping to execute a method (openApp('appName')) I have in the exposeInMainWorld that communicates with main.ts and creates a windows to open the app. 

Is it possible that from the wdio test I can execute openApp('appName') so it can then start the app I want without having any browser window opened?

If any details are needed I can provide more info.

Thanks in advance :)

**Comment by christian-bromann** [comment]
I am not sure where this `openApp` method comes from or how it is implemented but you can interact with all Electron APIs as part of your test. Note: if you open an non Electron app as part of your test case you won't be able to automate it since it is not your Electron application. For general desktop testing we recommend to take a look at: https://webdriver.io/docs/desktop-testing/macos

**Reply by MohammedChaibi** [reply]
Thank you for your reply!

What I'm trying to do is the following:
I have my preload.ts that has this
`contextBridge.exposeInMainWorld('electronApi', {
  openApp: async(appName:string) =>{
    ipcRenderer.invoke('openWindow',appName)
  }
});`

So when I'm in the browser that electron generates I can type `window.electronAPi.openApp(appName)` and this will load one of my apps. Basically using preload.ts to communicate the angular app with the electron side (I understand is the standard)

The testing of an app is already done using wdio with no problem, what I'm aiming to do is run this `electronAPi.openApp(appName)` but directly in the test of wdio and without the need of having a window opened because this is actually what is opening the window. Maybe I should do it in another way, my electron app has a menu tray and from there the user can select the app to open. The apps are just angular apps that I load with electron dynamically.

Is what I'm trying to do even possible? Is there a better way using wdio to achieve my goal?

Thanks again for the reply :)

**Reply by christian-bromann** [reply]
I am not sure if you can run the tests without having the Electron window being opened. Have you looked into the [docs](https://webdriver.io/docs/desktop-testing/electron/api) to use the `execute` command to hook into the Electron environment? I think you should be able to tigger the `openApp` call this way.

**Reply by MohammedChaibi** [reply]
I tried that, but it uses browser so I'm guessing it needs a browser to be opened, I will play with that a little bit to see if I can get it to work.
